ibm/service/power/resource_ibm_pi_network.go

```diff
@@ -200,18 +199,25 @@ func resourceIBMPINetworkCreate(ctx context.Context, d *schema.ResourceData, met
 
 		ipBodyRanges = []*models.IPAddressRange{{EndingIPAddress: &lastip, StartingIPAddress: &firstip}}
 
-		if g, ok := d.GetOk(helpers.PINetworkGateway); ok {
+		if g, ok := d.GetOk(Arg_Gateway); ok {
 			gateway = g.(string)
 		}
 
-		if ips, ok := d.GetOk(helpers.PINetworkIPAddressRange); ok {
+		if ips, ok := d.GetOk(Arg_IPAddressRange); ok {
 			ipBodyRanges = getIPAddressRanges(ips.([]interface{}))
 		}
 
 		body.IPAddressRanges = ipBodyRanges
 		body.Gateway = gateway
 		body.Cidr = networkcidr
 	}
+
+	if networktype == PubVlan {
+		if _, ok := d.GetOk(Arg_Cidr); ok {
+			return diag.Errorf("%s cannot be set when %s is dhcp-vlan or vlan", Arg_Cidr, Arg_NetworkType)
+		}
+	}
```

Comment:
```suggestion
	if _, ok := d.GetOk(Arg_Cidr); ok && networktype == PubVlan {
		return diag.Errorf("%s cannot be set when %s is dhcp-vlan or vlan", Arg_Cidr, Arg_NetworkType)
	}
```

Just a suggestion to avoid unnecessary nested if